OpenRA.Game/Map/MapCache.cs

```diff
@@ -174,7 +174,8 @@ void LoadAsyncInternal()
 					if (bitmap == null)
 					{
 						createdPreview = true;
-						bitmap = Minimap.RenderMapPreview(modData.DefaultRules.TileSets[p.Map.Tileset], p.Map, modData.DefaultRules, true);
+						var map = new Map(p.Path);
```

Comment:
You're generally only getting away with this because this thread loads maps one at a time. If it attempted to parallelize the loading you'd probably get way more random crashes.